Initial Solution Pack Merge

fortijames opened the pull request:
This is the first release of the solution pack for review. Going forward I'll continue to fork the ( https://github.com/fortinet-fortisoar/solution-pack-fortimanager-ztp-framework ) repo and push changes. I also intend to leverage ( https://github.com/fortinet-fortisoar/solution-pack-fortimanager-ztp-framework/issues ) for bugs and features.

cs-rohit-khune commented on connectors/data.json:
```diff
@@ -0,0 +1,44 @@
+                "name": "Default",
+                "default": true,
+                "config": {
+                    "allow_imports": true
```
@fortijames Configuring the code snippet connector exclusively with the libraries used in the content pack is advisable. Allowing all library imports is not recommended.

fortijames replied:
> @fortijames Configuring the code snippet connector exclusively with the libraries used in the content pack is advisable. Allowing all library imports is not recommended.

I just moved this to a setup requirement for now. We can cleanup later.

cs-rohit-khune commented on README.md:
```diff
@@ -1 +1,37 @@
-# solution-template-repository
\ No newline at end of file
+# Release Information
+ - **Version**: 1.0.3
```
@fortijames Please make sure to set the version to 1.0.0 and commit your solution pack code to the release/1.0.0 branch instead of the develop branch.

cs-rohit-khune replied:
@fortijames Please create documentation using the standard format provided in  https://github.com/fortinet-fortisoar/how-tos/tree/main/documenting

fortijames replied:
> @fortijames Please create documentation using the standard format provided in https://github.com/fortinet-fortisoar/how-tos/tree/main/documenting

Is there a tool for creating `contents.md`? That would be handy. :-)

cs-rohit-khune replied:
No, there is no tool available for this

cs-rohit-khune commented on connectors/data.json:
```diff
@@ -0,0 +1,44 @@
+        }
+    },
+    {
+        "name": "fortinet-fortimanager_dev",
```
@fortijames Please ensure that the released version of the connector is added. Instead of "fortinet-fortimanager_dev", please add "fortinet-fortimanager".

fortijames replied:
> @fortijames Please ensure that the released version of the connector is added. Instead of "fortinet-fortimanager_dev", please add "fortinet-fortimanager".

This one is pending a connector that works with the solution pack. You just created the one we will be using. 
https://github.com/fortinet-fortisoar/connector-fortinet-fortimanager-json-rpc

My thinking here was to get things rolling so we have something started and then push updates to fix the dev connector. Anyone that uses this solution pack as a pre-release would need to download the dev connector and I can put that in the setup docs. I mentioned this in email as a step required before publishing to the Content Hub. 

In short. I expect this to be changed in the next PR and did not expect it to stop the merge of the rest of the SP.

cs-rohit-khune commented on connectors/data.json:
```diff
@@ -0,0 +1,44 @@
+        "category": "Centralized Security Management",
+        "description": "The Fortinet FortiManager provides easy centralized configuration, policy-based provisioning, update management and end-to-end network monitoring for your Fortinet installed environment.",
+        "publisher": null,
+        "version": "3.0.0",
```
- do not mention the version in JSON, please remove this key

fortijames replied:
> * do not mention the version in JSON, please remove this key

This connector is in the process of being replaced.

cs-rohit-khune commented on README.md:
```diff
@@ -1 +1,37 @@
\ No newline at end of file
+# Release Information
+ - **Version**: 1.0.3
+ - **Certified**: unknown
```
Mark it as "No" instead of "unknown".

fortijames replied:
> Mark it as "No" instead of "unknown".

Ok, how do we get it Certified? Is there a doc on this?

cs-rohit-khune commented on dashboards/FortiManager Device Metafield Templates.json:
```diff
@@ -0,0 +1,105 @@
+    "displayName": "FortiManager Device Metafield Templates",
+    "type": "rows",
+    "templateType": "dashboard",
+    "roles": [],
```
@fortijames Please ensure that a role is assigned to the dashboard.

cs-rohit-khune commented on dashboards/FortiManager Metadata Assignments.json:
```diff
@@ -0,0 +1,15 @@
+    "displayName": "FortiManager Metadata Assignments",
+    "type": "rows",
+    "templateType": "dashboard",
+    "roles": [],
```
Please ensure that a role is assigned to the dashboard. @fortijames

cs-rohit-khune commented on dashboards/FortiManager Product Details.json:
```diff
@@ -0,0 +1,234 @@
+    "displayName": "FortiManager Product Details",
+    "type": "rows",
+    "templateType": "dashboard",
+    "roles": [],
```
@fortijames Please ensure that a role is assigned to the dashboard.

cs-rohit-khune commented on dashboards/FortiManager ZTP Profile and Phase Summary.json:
```diff
@@ -0,0 +1,385 @@
+    "displayName": "FortiManager ZTP Profile and Phase Summary",
+    "type": "rows",
+    "templateType": "dashboard",
+    "roles": [],
```
@fortijames Please ensure that a role is assigned to the dashboard.

cs-rohit-khune commented on dashboards/FortiManager ZTPF Docs - Metafield Sources.json:
```diff
@@ -0,0 +1,79 @@
+    "displayName": "FortiManager ZTPF Docs - Metafield Sources",
+    "type": "rows",
+    "templateType": "dashboard",
+    "roles": [],
```
Please ensure that a role is assigned to the dashboard.

cs-rohit-khune commented on dashboards/FortiManager ZTPF Docs - ZTP Profiles.json:
```diff
@@ -0,0 +1,79 @@
+    "displayName": "FortiManager ZTPF Docs - ZTP Profiles",
+    "type": "rows",
+    "templateType": "dashboard",
+    "roles": [],
```
@fortijames Please ensure that a role is assigned to the dashboard.

cs-rohit-khune commented on info.json:
```diff
@@ -0,0 +1,223 @@
+{
+    "name": "fortiManagerZTPFlow",
+    "version": "1.0.5",
```
@fortijames The version should be 1.0.0

cs-rohit-khune commented on info.json:
```diff
@@ -0,0 +1,223 @@
+    "type": "solutionpack",
+    "local": true,
+    "label": "FortiManager ZTP Flow",
+    "dependencies": [],
```
@fortijames Make sure to add any solution pack dependency such as "SOAR Framework"

cs-rohit-khune commented on info.json:
```diff
@@ -0,0 +1,223 @@
+    "label": "FortiManager ZTP Flow",
+    "dependencies": [],
+    "prerequisite": "# FortiSOAR System Settings\n - Modify the `System Settings>Appliances>Playbook` user to have the `FortiManager-Playbook` Role. \n - Modify `System Settings>Users`, that will be using this solution, to have the `FortiManager-Admin` Role. \n\n# FortiManager API User\nAdd a FortiSOAR API user to your FMG that has RPC read-write settings. Below is an example user that includes a ...
+    "publisher": "Fortinet",
```
@fortijames publisher should be "Fortinet CSE"

cs-rohit-khune commented on info.json:
```diff
@@ -0,0 +1,223 @@
+    "prerequisite": "# FortiSOAR System Settings\n - Modify the `System Settings>Appliances>Playbook` user to have the `FortiManager-Playbook` Role. \n - Modify `System Settings>Users`, that will be using this solution, to have the `FortiManager-Admin` Role. \n\n# FortiManager API User\nAdd a FortiSOAR API user to your FMG that has RPC read-write settings. Below is an example user that includes a ...
+    "publisher": "Fortinet",
+    "description": "A solution framework to manage one or more FortiManagers and attached Fortinet Devices. Build a ZTP Flow that will perform onboarding, lifecycle, and/or offboarding objectives for devices. Create a single pane of glass dashboard for multiple FortiManagers and attached devices. Create reports for devices using jinja to simplify data collection. Interact with external users and/ ...
+    "help": "https://github.com/fortinet-fortisoar/solution-pack-fortimanager-ztp-framework",
```
Please provide a link to README.md on the release/1.0.0 branch. i.e

https://github.com/fortinet-fortisoar/solution-pack-fortimanager-ztp-framework/blob/release/1.0.0/README.md

fortijames replied:
> Please provide a link to README.md on the release/1.0.0 branch. i.e
> 
> https://github.com/fortinet-fortisoar/solution-pack-fortimanager-ztp-framework/blob/release/1.0.0/README.md

Since I don't know the best path here. I will cancel my PR and redo as a 1.0.0 branch request.

cs-rohit-khune commented on info.json:
```diff
@@ -0,0 +1,223 @@
+        "globalVariables": [
+            {
+                "id": 7,
+                "name": "Server_fqhn",
```
@fortijames Please remove the global variable 'Server_fqdn' as its value may vary across different FSR instances.

fortijames replied:
> @fortijames Please remove the global variable 'Server_fqdn' as its value may vary across different FSR instances.

I don't know how this is being exported. I have Global Vars unchecked.

cs-rohit-khune commented on info.json:
```diff
@@ -0,0 +1,223 @@
+                "apiName": "code-snippet"
+            },
+            {
+                "name": "Fortinet FortiManager Custom",
```
@fortijames please mention released released connector here

fortijames replied:
> @fortijames please mention released released connector here

I assume you mean the `Fortinet FortiManager Custom`. This connector is being replaced.

cs-rohit-khune commented on roles/FortiManager-Admin-c57340f1-c387-474b-9c24-effc9e3c41aa.json:
```diff
@@ -0,0 +1,585 @@
+    "@type": "Role",
+    "description": "The FortiManager Admin should be able to edit only certain fields as the synchronization process owns many fields.",
+    "name": "FortiManager-Admin",
+    "modulePermissions": [
```
@fortijames Please include only the custom modules that are added with this solution pack in the role. Exclude Out-of-the-Box (OOB) modules such as agents, alerts, and so on.

cs-rohit-khune commented on roles/FortiManager-Playbook-c064238d-8049-4c5b-b3e8-e3a1c41178fe.json:
```diff
@@ -0,0 +1,531 @@
+{
+    "@context": "\/api\/3\/contexts\/Role",
```
@fortijames Please include only the custom modules that are added with this solution pack in the role. Exclude Out-of-the-Box (OOB) modules such as agents, alerts, and so on.

cs-rohit-khune commented on playbooks/FortiManager(FMG)-Triggers/On ZTP Profile Assignment in Device Record.json:
```diff
@@ -0,0 +1,557 @@
+{
+    "@type": "Workflow",
+    "triggerLimit": null,
+    "name": "On ZTP Profile Assignment in Device Record",
```
@fortijames Trigger condition is missing in trigger step 

<img width="565" alt="image" src="https://github.com/fortinet-fortisoar/solution-pack-fortimanager-ztp-framework/assets/46894140/a9059c3d-6ee4-4c87-ad35-b5869dbb503c">

cs-rohit-khune commented:
@fortijames Although already mentioned in the pre-requisite instructions, it is recommended to add a note in the documentation that instructs users to assign the 'FortiManager-Playbook' and 'FortiManager-Admin' roles.

fortijames commented:
What's the process for addressing comments in the PR? 
 - Is there a way to update my PR code with my latest changes? I did not see that option. 
 - Close this PR and create a new? 
 - Create a new PR with additions and have you merge/play them in order?

fortijames commented:
I'm closing and will try again. Please let me know the proper flow for making changes within a PR.
